_posts/2015-10-12-creating-bem-stylesheets-for-react-components.markdown

```diff
@@ -0,0 +1,279 @@
+
+I was recently creating a new React component for our "Book a Tour" modal, called LocationCmInfo ("Cm" standing for Community Manager, the person giving the tour.)
+
+
+```js
+import React, {Component, PropTypes} from 'react';
+
+class LocationCmInfo extends Component {
+  render() {
+    const { actions, selectedLocation } = this.props;
+
+    return (
+      <div className="modal__content clearfix">
+        <div className="location-cm-info">
+          <div className="location-cm-info__content">
+            <div className="modal__exit" onClick={actions.modalsFlush}>&#215;</div>
+            <div className="location-cm-info__image">
+              <img src={selectedLocation.community_manager.image_url} />
+            </div>
+            <div className="location-cm-info__text">
+              <h1 className="apercu">Come visit us!</h1>
+              <h2 className="apercu">{selectedLocation.name} - {selectedLocation.line1}</h2>
+              <p>{selectedLocation.community_manager.name} is a Community Manager at {selectedLocation.name} and will give you a tour of the building.</p>
+            </div>
+          </div>
+        </div>
+      </div>
+    )
+  }
+}
+
+LocationCmInfo.propTypes = {
+  actions: PropTypes.object.isRequired,
+  selectedLocation: PropTypes.object.isRequired
+};
+
+export default LocationCmInfo;
+```
+
+The markup is incredibly readable and easy to comprehend. We have a block (`.location-cm-info`) that contains an element (`.location-cm-info__content`) containing 2 other elements (`.location-cm-info__image` and `.location-cm-info__text`.) There are no modifiers yet.
+
+I was taking a look at this the other day and realized that **this CSS could use a little refactoring**:
+
+```sass
+.location-cm-info {
+  &__content {
+    position: relative;
+    background-color: $syracuse;
+    @media #{$medium-up} { padding: 45px; }
+    @media #{$small-only} { padding: 30px 25px; }
+  }
+
+  &__image {
+    border-radius: 50%;
+    overflow: hidden;
+
+    @media #{$medium-up} {
+      position: absolute;
+      top: 45px;
+      left: 30px;
+    }
+    @media #{$small-only} {
+      position: relative;
+      text-align: center;
+      margin: 0 0 15px 0;
+    }
+
+    img { width: 100px; height: 100px; }
+  }
+
+  &__text {
+
+    @media #{$medium-up} { margin: 0 0 0 120px; }
+
+    @media #{$small-only} { text-align: center; }
+
+    h1, h2 { line-height: 1; }
+
+    h1 {
+      font-size: 25px;
+      margin: 0 0 12px 0;
+    }
+    h2 {
+      font-size: 15px;
+      margin: 0 0 14px 0;
+    }
+    p { font-size: 11px; margin: 0; }
+  }
+}
+```
+
+I've been breaking my own rule: "Do not use HTML elements in selectors."  Let's first refactor the markup and change the stylesheet accordingly. First of all, I would like to rename this component `.location-info` and add a modifier `--with-cm` to clarify that this module can be used for more than just highlighting a Community Manager.
```

Comment:
same thing as above, for the purposes of the tutorial maybe have it say "--with-image" instead of "--with-cm" to make it more readable to peeps who don't know about our community managers.